plugin/src/test/java/org/wildfly/plugins/bootablejar/maven/goals/DevWatchContextTestCase.java

```diff
@@ -0,0 +1,726 @@
+            Files.write(project.pomFile, expected.getBytes());
+            BootableAppEventHandler handler = checkEvent(ctx, projCtx, false, false, true, false, false, new Checker() {
+                @Override
+                public boolean isOk(BootableAppEventHandler handler) {
+                    return handler.seenUpdated.size() == 1 && handler.deleted.size() == 1;
+                }
+            });
+            Assert.assertTrue(handler.reset);
+            Assert.assertFalse(handler.rebuildBootableJAR);
+            Assert.assertTrue(handler.copied.isEmpty());
+            Assert.assertEquals(1, handler.deleted.size());
+            Assert.assertEquals(1, handler.seenUpdated.size());
+            Assert.assertTrue(handler.seenUpdated.contains(project.pomFile));
+            Assert.assertEquals(project.pomFile, handler.deleted.get(project.pomFile));
+            projCtx.reset();
+
+            // do a change that would imply a full rebuild
+            projCtx.pluginConfigUpdated = true;
+            Files.delete(project.pomFile);
+            Files.createFile(project.pomFile);
+            expected = "<!-- An empty pom file, updated by test with change to the plugin config-->";
+            Files.write(project.pomFile, expected.getBytes());
+            handler = checkEvent(ctx, projCtx, false, false, true, false, false, new Checker() {
+                @Override
+                public boolean isOk(BootableAppEventHandler handler) {
+                    return handler.seenUpdated.size() == 1;
+                }
+            });
+            Assert.assertTrue(handler.rebuildBootableJAR);
+            Assert.assertFalse(handler.reset);
+            Assert.assertTrue(handler.copied.isEmpty());
+            Assert.assertEquals(1, handler.seenUpdated.size());
+            Assert.assertTrue(handler.seenUpdated.contains(project.pomFile));
+            projCtx.reset();
+
+        } finally {
+            if (ctx != null) {
+                ctx.cleanup();
+            }
+            project.cleanup();
+        }
+    }
+
+    private BootableAppEventHandler checkEvent(DevWatchContext ctx, TestProjectContext projCtx,
+            boolean cleanup, boolean compile, boolean checkPluginConfig,
+            boolean packageJar, boolean packageWar, Checker checker) throws Exception {
+
+        boolean ok = false;
+        BootableAppEventHandler handler = null;
+        WatchKey key = null;
+        long timeout = 60;
+        do {
+            key = watcher.poll();
+            if (key != null) {
+                if (handler == null) {
+                    handler = ctx.newEventHandler();
+                }
+                for (WatchEvent<?> event : key.pollEvents()) {
+                    @SuppressWarnings("unchecked")
+                    WatchEvent<Path> ev = (WatchEvent<Path>) event;
+                    System.out.println("New event " + ev.kind() + " path " + ev.context());
```

Comment:
Probably not a huge deal, but this could be a bit verbose.

Reply:
What I could do is store these traces in memory and only dump them to console if we have an exception?

Reply:
The more I think about it, it's likely okay. The other option would be to use a logger too. Either way works and we definitely shouldn't block merging this based on this.

Reply:
I am redirecting the surefire output to a file. It is interesting to have these traces when the test fail in GitHub action jobs. I introduced a new property to disable this redirect to file.